src/Http/Routing/src/ShortCircuit/RouteShortCircuitEndpointConventionBuilderExtensions.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Routing.ShortCircuit;
+
+namespace Microsoft.AspNetCore.Builder;
+
+/// <summary>
+/// 
+/// </summary>
+public static class RouteShortCircuitEndpointConventionBuilderExtensions
+{
+    private static readonly ShortCircuitMetadata _nullShortCircuitMetadata = new ShortCircuitMetadata(null);
```

Comment:
Does it worth to also add a dictionary with different status codes in order to reuse the metadata?

Reply:
Maybe just statics and a switch statement for the most common? null, 200, 404, 401.